src/index.js

```diff
@@ -53,105 +48,65 @@ function formatSize(size, filename, type, raw) {
 	)}: ${chalk.white(basename(filename))}.${type}`;
 }
 
-export default async function microbundle(options) {
-	let cwd = (options.cwd = resolve(process.cwd(), options.cwd)),
-		hasPackageJson = true;
+export default async function microbundle(inputOptions) {
+	let options = { ...inputOptions };
 
-	try {
-		options.pkg = JSON.parse(
-			await readFile(resolve(cwd, 'package.json'), 'utf8'),
-		);
-	} catch (err) {
-		stderr(
-			chalk.yellow(
-				`${chalk.yellow.inverse(
-					'WARN',
-				)} no package.json found. Assuming a pkg.name of "${basename(
-					options.cwd,
-				)}".`,
-			),
-		);
-		let msg = String(err.message || err);
-		if (!msg.match(/ENOENT/)) stderr(`  ${chalk.red.dim(msg)}`);
-		options.pkg = {};
-		hasPackageJson = false;
-	}
+	options.cwd = resolve(process.cwd(), inputOptions.cwd);
+	const cwd = options.cwd;
 
-	if (!options.pkg.name) {
-		options.pkg.name = basename(options.cwd);
-		if (hasPackageJson) {
-			stderr(
-				chalk.yellow(
-					`${chalk.yellow.inverse(
-						'WARN',
-					)} missing package.json "name" field. Assuming "${
-						options.pkg.name
-					}".`,
-				),
-			);
-		}
-	}
+	const { hasPackageJson, pkg } = await getConfigFromPkgJson(cwd);
+	options.pkg = pkg;
 
-	options.name =
-		options.name || options.pkg.amdName || safeVariableName(options.pkg.name);
+	const { finalName, pkgName } = getName({
+		name: options.name,
+		pkgName: options.pkg.name,
+		amdName: options.pkg.amdName,
+		hasPackageJson,
+		cwd,
+	});
+
+	options.name = finalName;
+	options.pkg.name = pkgName;
 
 	if (options.sourcemap !== false) {
 		options.sourcemap = true;
 	}
 
-	const jsOrTs = async filename =>
-		resolve(
-			cwd,
-			`${filename}${
-				(await isFile(resolve(cwd, filename + '.ts')))
-					? '.ts'
-					: (await isFile(resolve(cwd, filename + '.tsx')))
-					? '.tsx'
-					: '.js'
-			}`,
-		);
-
-	options.input = [];
-	[]
-		.concat(
-			options.entries && options.entries.length
-				? options.entries
-				: (options.pkg.source && resolve(cwd, options.pkg.source)) ||
-						((await isDir(resolve(cwd, 'src'))) &&
-							(await jsOrTs('src/index'))) ||
-						(await jsOrTs('index')) ||
-						options.pkg.module,
-		)
-		.map(file => glob(file))
-		.forEach(file => options.input.push(...file));
-
-	let main = resolve(cwd, options.output || options.pkg.main || 'dist');
-	if (!main.match(/\.[a-z]+$/) || (await isDir(main))) {
-		main = resolve(main, `${removeScope(options.pkg.name)}.js`);
-	}
-	options.output = main;
+	options.input = await getInput({
+		entries: options.entries,
+		cwd,
+		source: options.pkg.source,
+		module: options.pkg.module,
```

Comment:
I’m curious to learn, unless this is a mistake; how is `module` used as an input here?